tasks/main.yml

```
@@ -40,6 +40,21 @@
     # see (https://docs.ansible.com/ansible-lint/rules/default_rules.html)
     - __network_package_install.changed  # noqa 503
 
+# If network packages have changed and team connections are required,
+# NetworkManager must be restarted
+- name: Restart NetworkManager due to team interfaces
+  service:
+    name: NetworkManager
+    state: restarted
+  when:
+    - __network_team_connections_defined
```

```suggestion
    - __network_wireless_connections_defined or __network_team_connections_defined
```

can be used in the first restart task to avoid repeating most common content and restarting NM twice in case both wireless and team interfaces are specified.

---

updated